test/jdk/java/lang/management/ThreadMXBean/Locks.java

```diff
@@ -61,10 +64,12 @@ private static void assertNoLock(Thread t) {
             return;
         }
         String name = t.getName();
+        // If t is Virtual, it will hold a lock on a VirtualThread, which we ignore.
         Optional<ThreadInfo> result = Arrays.stream(
                 TM.getThreadInfo(TM.getAllThreadIds(), true, true))
                                             .filter(Objects::nonNull)
                                             .filter(i -> name.equals(i.getLockOwnerName()))
+                                            .filter(i  -> !i.getLockName().contains("java.lang.VirtualThread"))
```

Comment:
```suggestion
                                            .filter(i -> !i.getLockName().contains("java.lang.VirtualThread"))
```

Reply:
ThreadMXBean::getAllThreadsIds returns an array of the thread IDs of platform threads, it doesn't include virtual threads, so I don't know why this filter is added.

Reply:
Whitespace correction, thanks.

Reply:
The filter was because of a failure where it's called on a real thread that happens to own a VirtualThread lock, like:

"ForkJoinPool-1-worker-1" daemon prio=5 Id=32 WAITING on java.lang.VirtualThread@289cb0b4 owned by "MainThread" Id=31

...actually my comment at line 67 isn't exactly correct, will update it.